### PR TITLE
ajax request for ordering panels is send on after finished resizing only

### PR DESCRIPTION
**Pull request by dcrystalj**
#284

### CODE REVIEW

**Comment by dcrystalj** [comment]
fixed only first issue 
- Ajax requests to store their position is send many times while window resizing, it should be send only at the end

File: piplmesh/frontend/static/piplmesh/js/home.js [hunk 1]
```diff
@@ -216,10 +216,7 @@ $(document).ready(function () {
         });
     });
 
-    // TODO: Ajax request to store panels state is currently send many times while resizing, it should be send only at the end
-    $(window).resize(function (event) {
-        initializePanels();
-    });
+    $(window).resize( $.debounce( 250, initializePanels) );
```
Comment (mitar) [inline piplmesh/frontend/static/piplmesh/js/home.js]: Looks good. But there is unnecessary space around.

Comment (mitar) [inline piplmesh/frontend/static/piplmesh/js/home.js]: Hmm. But be careful. We probably do want that panels are repositioned automatically while window is resizing? Only we should not be doing Ajax requests to store the state until the very end.

**Comment by mitar** [comment]
So this is a good idea. To use a library and to simply wrap in such a way. But you should do this only for Ajax requests.

**Comment by dcrystalj** [comment]
Is not as easy as it might looks like. 
It's first time for me doing javascript =)

File: piplmesh/frontend/static/piplmesh/js/home.js [hunk 2]
```diff
@@ -281,9 +281,10 @@ $(document).ready(function () {
         });
     });
 
-    // TODO: Ajax request to store panels state is currently send many times while resizing, it should be send only at the end
+    $(window).resize($.debounce(250, initializePanels));
     $(window).resize(function (event) {
-        initializePanels();
+        initializeEmptyColumnsForPanels();
+        orderPanelsDefault();
```
Comment (mitar) [inline piplmesh/frontend/static/piplmesh/js/home.js:287]: I think you also have to call `makeColumnsSortable`.

And `orderPanelsDefault` is not really the correct thing, no? It reorders thing.

**Comment by mitar** [comment]
So this is not so easy. :-)

So when you are resizing, you should in fact reorder them properly when window becomes wide enough for another column. And the way you reorder that should be the same as it is stored in the database. So then you have to make an Ajax request. But just when number of columns change.

It seems it does not store anything at the end of the resizing.

So the problem is in fact just that it reads from the server too many times, not just when number of columns change. Sorry for confusion.

**Comment by mitar** [comment]
Do you want to work more on this?

**Comment by dcrystalj** [comment]
Not really, don't have time.

**Pull request closed without merging**